localstack/services/awslambda/event_source_listeners/sqs_event_source_listener.py

```diff
@@ -59,21 +57,17 @@ def _listener_loop(self, *args):
 
                     try:
                         queue_url = aws_stack.sqs_queue_url_for_arn(queue_arn)
-                        messages = unprocessed_messages.pop(queue_arn, None)
+                        result = sqs_client.receive_message(
+                            QueueUrl=queue_url,
+                            AttributeNames=["All"],
+                            MessageAttributeNames=["All"],
+                            MaxNumberOfMessages=batch_size,
+                        )
```

Comment:
totally agree. unfortunately we can't block here because of the way the loop is structured. see comment here: https://github.com/localstack/localstack/issues/5042#issuecomment-1207394650

that would require a thread per event-source-mapping, which again requires a different interface for the event source listener (adding/removing mappings). i'd love to do that as part of the lambda rework.